### PR TITLE
🧅 Add Cloudflare DNS onion protocol

### LINKED ISSUE #239: 🧅 add Onion protocol to known providers
## Description

add Onion protocol to known DNS resolvers

**URL of affected page:**
http://www.xpqizwv63kdv3wxcecdrokx24itbts7y2tedemntqeyc2wbm3lrobbid.onion/providers/dns/

### PR DESCRIPTION
**Pull request by 4-FLOSS-Free-Libre-Open-Source-Software**
## Description

Resolves #239

This is no recommendation of using cloudflare, just existing protocol list addition.
 Source: [(DNS over TCP, TLS, and HTTPS) over Onion](https://developers.cloudflare.com/1.1.1.1/other-ways-to-use-1.1.1.1/dns-over-tor#dns-over-tcp-tls-and-https)

https://dns4torpnlfs2ifuz2s2yf3fc7rdmsbhm6rw75euj35pac6ap25zgqad.onion/dns-query
tor.cloudflare-dns.com

### CODE REVIEW

File: _data/dns/cloudflare.yml [hunk 1]
```diff
@@ -16,6 +16,7 @@ logs:
     The 1.1.1.1 resolver service does not log personal data,
     and the bulk of the limited non-personally identifiable query data is only stored for 25 hours."
 protocols:
+  - name: Onion
```
Comment (rusty-snake) [inline _data/dns/cloudflare.yml:19]: Usually this is called `DNS over Tor` (and `DNS over HTTPS over Tor`, etc.).

https://developers.cloudflare.com/1.1.1.1/other-ways-to-use-1.1.1.1/dns-over-tor

Comment (4-FLOSS-Free-Libre-Open-Source-Software) [reply _data/dns/cloudflare.yml:19]: > 
> 
> Usually this is called `DNS over Tor` 

`DoO` 
`DoHoO` ?

Comment (dngray) [reply _data/dns/cloudflare.yml]: My understanding is DNS over Tor is pretty uncommon, as if you're going to that extent why not just use Tor.

Comment (4-FLOSS-Free-Libre-Open-Source-Software) [reply _data/dns/cloudflare.yml]: Tor missing a full DNSstack and its current DNSPort implentation does not support anything else than requests about `AAAA`, `A` and `PTR` requests.

Comment (dngray) [reply _data/dns/cloudflare.yml]: I actually wonder how much longer that will be the case given HTTP/3 (QUIC) is UDP

**Comment by jonaharagon** [comment]
As @dngray mentioned, this is still marked as experimental. The page that outlines 1.1.1.1 over Tor also lists DNS over Discord, DNS over Telegram, and DNS over Google Sheets as 1.1.1.1 resolvers, but we are not including those here. I would lean towards not mentioning these at all and closing this PR.

**Pull request closed without merging**